web/src/hooks/useApps.ts

```diff
@@ -19,140 +19,299 @@ enum DescriptionSubFields {
     $is_developer_allow_listing: Boolean
-    $world_app_description: String = ""
+    $integration_url: String = ""
+    $app_website_url: String = ""
+    $source_code_url: String = ""
   ) {
-    update_app_by_pk(
-      pk_columns: { id: $id }
-      _set: {
-        status: $status
+    insert_app_metadata_one(
+      object: {
+        app_id: $app_id
         name: $name
-        description_internal: $description_internal
-        link_to_integration: $link_to_integration
+        logo_img_url: $logo_img_url
+        showcase_img_urls: $showcase_img_urls
+        hero_image_url: $hero_image_url
+        description: $description
+        world_app_description: $world_app_description
         category: $category
         is_developer_allow_listing: $is_developer_allow_listing
-        world_app_description: $world_app_description
+        integration_url: $integration_url
+        app_website_url: $app_website_url
+        source_code_url: $source_code_url
+      }
+      on_conflict: {
+        constraint: app_metadata_unique_verification_status_row_key
+        update_columns: [
+          name
+          logo_img_url
+          showcase_img_urls
+          hero_image_url
+          description
+          world_app_description
+          category
+          is_developer_allow_listing
+          integration_url
+          app_website_url
+          source_code_url
+        ]
+        where: {
+          status: { _neq: "verified" }
+        }
       }
     ) {
-      ${appFields}
+      ${appMetadataFields}
     }
   }
 `;
 
 const InsertAppQuery = gql`
-  mutation InsertApp($object: app_insert_input!) {
-    insert_app_one(object: $object) {
+  mutation InsertApp($appObject: app_insert_input!) {
+    insert_app_one(object: $appObject) {
       ${appFields}
     }
   }
 `;
 
+const InsertAppMetadataQuery = gql`
+  mutation InsertAppMetadata($appMetadataObject: app_metadata_insert_input!) {
+    insert_app_metadata_one(object:$appMetadataObject) {
+      ${appMetadataFields}
+    }
+  }
+`;
+
 const DeleteAppQuery = gql`
   mutation DeleteApp($id: String!) {
-    app: delete_app_by_pk(id: $id) {
+    delete_app_by_pk(id: $id) {
       id
     }
   }
 `;
-
-const fetchApps = async () => {
+const fetchApps = async (
+  attemptedToHandleMissingMetadata: boolean = false
+): Promise<Array<AppModel>> => {
   const response = await graphQLRequest<{
     app: Array<AppModel>;
   }>({
     query: FetchAppsQuery,
   });
 
-  if (response.data?.app.length) {
-    return response.data.app;
+  const apps = response.data?.app || [];
+  // Even verified apps will have an unverified row. Thus the only case this will be true is if we have an orphaned app row
+  const appsMissingMetadata = apps.filter((app) => !app.app_metadata);
+
+  // Checks for orphaned app rows and tries to add a metadata row
+  if (appsMissingMetadata.length > 0 && !attemptedToHandleMissingMetadata) {
+    await _handleMissingMetadata(appsMissingMetadata);
```

Comment:
doing this when fetching apps is mixing concerns, perhaps best to do this when doing updates?